Diff/diff.py

```diff
@@ -45,120 +56,176 @@ def run(self, files):
             return
 
         try:
-            a = read_file_lines(files[1])
-            b = read_file_lines(files[0])
+            a: typing.List[str] = read_file_lines(files[1])
+            b: typing.List[str] = read_file_lines(files[0])
         except UnicodeDecodeError:
             sublime.status_message("Diff only works with UTF-8 files")
             return
 
-        adate = time.ctime(os.stat(files[1]).st_mtime)
-        bdate = time.ctime(os.stat(files[0]).st_mtime)
+        adate: str = time.ctime(os.stat(files[1]).st_mtime)
+        bdate: str = time.ctime(os.stat(files[0]).st_mtime)
 
-        diff = difflib.unified_diff(a, b, files[1], files[0], adate, bdate)
-        show_diff_output(diff, None, self.window, f"{os.path.basename(files[1])} -> {os.path.basename(files[0])}", 'diff_files', 'diff_files_to_buffer')
+        diff = difflib.unified_diff(
+            a,
+            b,
+            files[1],
+            files[0],
+            adate,
+            bdate
+        )
+        show_diff_output(
+            diff,
+            None,
+            self.window,
+            f"{os.path.basename(files[1])} -> {os.path.basename(files[0])}",
+            'diff_files',
+            'diff_files_to_buffer'
+        )
 
     def is_visible(self, files):
         return len(files) == 2
 
 
 class DiffChangesCommand(sublime_plugin.TextCommand):
 
-    def run(self, edit):
+    def run(self, edit: sublime.Edit):
+
+        v: sublime.View = self.view
+        w: typing.Union[None, sublime.Window] = v.window()
+        if not w:
+            sublime.status_message('Unable to diff with missing window')
+            return
 
-        fname = self.view.file_name()
+        fname: typing.Union[None, str] = v.file_name()
```

Comment:
`typing.Optional` is more appropriate in those cases. But also there is no point in annotating types in those cases since the type will be evaluated from the value already.